clock.c

```diff
@@ -311,7 +311,7 @@ static void clock_prune_subscriptions(struct clock *c)
 
 	clock_gettime(CLOCK_MONOTONIC, &now);
 	LIST_FOREACH_SAFE(s, &c->subscribers, list, tmp) {
-		if (s->expiration <= now.tv_sec) {
+		if (s->expiration != 0 && s->expiration <= now.tv_sec) {
```

Comment:
put a comment/note here saying The subscription will remain active if the subscription duration is set to zero.